### ISSUE TITLE
The Sign up button from the Connect with us section doesn't respond to mouse hovers

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons#7072]
Follow up for mozilla/addons#7044 

### Describe the problem and steps to reproduce it:
1. Open DevHub homepage in logged out view - https://addons.allizom.org/en-US/developers/
2. Scroll down to the Connect with us section (blue footer)
3. Hover over the Sign up button form the Extensions Developer Newsletter component

### What happened?
There is no response from the button on mouse hover (mouse pointer doesn't change)

### What did you expect to happen?
The pointer should change when hovering over the button to indicate an interaction

### Anything else we should know?
- I've added a comparison with Extension workshop to illustrate what the expected result should be

[DevHub]
![2019-10-23_09-20-27](https://user-images.githubusercontent.com/31961530/67364729-cca8b580-f578-11e9-89e8-5026fc11a8ab.gif)

[Extension workshop]
![2019-10-23_09-21-05](https://user-images.githubusercontent.com/31961530/67364739-d3372d00-f578-11e9-8177-b47afe85059b.gif)
